figanos/matplotlib/plot.py

```diff
@@ -1543,9 +1547,10 @@ def scattermap(
         else:
             raise ValueError("If `data` is a dict, it must be of length 1.")
 
-    # select data to plot
+    # select data to plot and its xr.Dataset
     if isinstance(data, xr.DataArray):
         plot_data = data
+        data = xr.Dataset({plot_data.name: plot_data})
```

Comment:
why does it need to be a dataset ? 
what if it doesn't have a name ?

Reply:
TBH, it's been a while that I added that so don't quite remember 🤔 ...

It was something missing on the plot if the dataset was not there.

Reply:
haha ok!